drivers/w3s.go

```diff
@@ -0,0 +1,340 @@
+package drivers
+
+import (
+	"context"
+	"fmt"
+	bserv "github.com/ipfs/go-blockservice"
+	"github.com/ipfs/go-cid"
+	ds "github.com/ipfs/go-datastore"
+	dssync "github.com/ipfs/go-datastore/sync"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	format "github.com/ipfs/go-ipld-format"
+	"github.com/ipfs/go-merkledag"
+	"github.com/ipfs/go-unixfs"
+	"github.com/ipld/go-car"
+	"io"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+)
+
+const w3SDefaultSaveTimeout = 5 * time.Minute
+
+var cidV1 = merkledag.V1CidPrefix()
+
+// This represents the main CAR directory structure organized by pubId.
+// Data for each pubId is removed after the CAR directory is published.
+var (
+	dataToPublish   = make(map[string]*rootCar)
```

Comment:
Yes, I think it could cause the memory leak if the `Publish()` is never called. The best solution would actually be to move this global variable into the session, then it would be automatically garbage collected. Let's do it as part of #16 

For now, we could add some clean-up with timeout. Wdyt?
On the other hand, maybe we can just leave it as it is. I think that if `Publish()` is not called, then it means we'll have some other issues in `catalyst-api`. And the worst thing that can actually happen is that Catalyst replica gets restarted.

Reply:
I had a second thought on that. IMO we should leave it as it is for now. Just treat as a hack using `pubId` and storing everything in the global variable. Then, after we go live with web3.storage, prioritize  #16. This will fix this issue anyway.

Also, FYI, here's the second part of this hack in the catalyst-api: https://github.com/livepeer/catalyst-api/commit/c8fe7f9aa7f3c99a8bb04fc645e42e3836afc615

Reply:
That sounds reasonable. WDYT about adding a comment to document that a memory leak is possible if `Publish()` is not called for some reason?

Reply:
Ok, added.